pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -166,8 +165,8 @@ func (pl *VolumeBinding) PreFilter(ctx context.Context, state *framework.CycleSt
 	if hasPVC, err := pl.podHasPVCs(pod); err != nil {
 		return nil, framework.NewStatus(framework.UnschedulableAndUnresolvable, err.Error())
 	} else if !hasPVC {
-		state.Write(stateKey, &stateData{skip: true})
-		return nil, nil
+		state.Write(stateKey, &stateData{})
```

Comment:
Why do we need to write state when returning Skip?

Reply:
We'll try to getStateData in `Reserve`, if we don't write state here, reserve will treat no state data as a no founding error.
[logic](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/volumebinding/volume_binding.go#L308)

Reply:
Ah, I see.
Hmm, I'd say we can just ignore the error from `getStateData` in `Reserve` when [cyclestate.SkipFilterPlugins](https://github.com/kubernetes/kubernetes/blob/3aff1f97bef13f1d40e47fae785994d57c3e5a80/pkg/scheduler/framework/cycle_state.go#L54) has `VolumeBinding` plugin.

Reply:
Here are two concerns:

1.  If we just ignore the error from `getStateData` in Reserve,  we'll still need to constrcut a `StateData` with `allBound` field and write it back to the cycleState. [logic](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/volumebinding/volume_binding.go#L319). So I don't think there's too much difference.
2. `getStateData` is used in many stages:`Filter`,`PreBind`,`Reserve`,`Score`,`Unreserve`.If we ignore this err in `Reserve`, we'll have to check `SkipFilterPlugins` in other stages too. There will be more code changes involved.

WDYT?

Reply:
> getStateData is used in many stages:Filter,PreBind,Reserve,Score,Unreserve.If we ignore this err in Reserve, we'll have to check SkipFilterPlugins in other stages too.

That makes sense. Ok, let's just keep it as it is.